File: source/com/barrybecker4/simulation/fluid/rendering/RenderingOptions.java
/** Copyright by Barry G. Becker, 2000-2011. Licensed under MIT License: http://www.opensource.org/licenses/MIT  */
package com.barrybecker4.simulation.fluid.rendering;

import com.barrybecker4.common.concurrency.*;


/**
 * Fluid rendering options
 *
 * @author Barry Becker
 */
public class RenderingOptions {

    /** scales the size of everything   */
    private static final double DEFAULT_SCALE = 4;
    private double scale = DEFAULT_SCALE;

    private boolean showVelocities = false;
    private boolean showPressures = true;
    private boolean useLinearInterpolation = false;
    private boolean showGrid = false;

    /** Manages the worker threads. */
    private RunnableParallelizer parallelizer;


    /**
     * Constructor
     */
    public RenderingOptions() {
        setParallelized(false);
        setScale(DEFAULT_SCALE);
    }

    public void setParallelized(boolean useParallelization) {

        parallelizer =
             useParallelization ? new RunnableParallelizer() : new RunnableParallelizer(1);
    }

    public boolean isParallelized() {
        return parallelizer.getNumThreads() > 1;
    }

    RunnableParallelizer getParallelizer() {
        return parallelizer;
    }

    public void setScale(double scale) {
        this.scale = scale;
    }

    public double getScale() {
        return scale;
    }

    public void setShowVelocities(boolean show) {
        showVelocities = show;
    }

    public boolean getShowVelocities() {
        return showVelocities;
    }

    public void setShowPressures(boolean show) {
        showPressures = show;
    }

    public boolean getShowPressures() {
        return showPressures;
    }

    public void setUseLinearInterpolation(boolean useInterp) {
        useLinearInterpolation = useInterp;
    }
    public boolean getUseLinearInterpolation() {
        return useLinearInterpolation;
    }

    public boolean getShowGrid() {
        return showGrid;
    }

    public void setShowGrid(boolean showGrid) {
        this.showGrid = showGrid;
    }
}

File: scala-source/com/barrybecker4/simulation/fluid/rendering/RenderingOptions.scala
// Copyright by Barry G. Becker, 2016-2017. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.simulation.fluid.rendering

import com.barrybecker4.common.concurrency._


/**
  * Fluid rendering options
  * @author Barry Becker
  */
object RenderingOptions {
  /** scales the size of everything. Number of pixels on edge of cell.   */
  private val DEFAULT_SCALE = 4
}

class RenderingOptions() {

  var isParallelized = true

  private var scale: Double = RenderingOptions.DEFAULT_SCALE
  private var showVelocities = false
  private var showPressures = true
  private var useLinearInterpolation = true
  private var showGrid = false

  def setScale(scale: Double): Unit = {this.scale = scale }
  def getScale: Double = scale

  def setShowVelocities(show: Boolean): Unit = { showVelocities = show}
  def getShowVelocities: Boolean = showVelocities

  def setShowPressures(show: Boolean): Unit = { showPressures = show }
  def getShowPressures: Boolean = showPressures

  def setUseLinearInterpolation(useInterp: Boolean): Unit = { useLinearInterpolation = useInterp}
  def getUseLinearInterpolation: Boolean = useLinearInterpolation

  def getShowGrid: Boolean = showGrid
  def setShowGrid(showGrid: Boolean): Unit = { this.showGrid = showGrid }
}
